tests/testsuite/version.rs

```diff
@@ -1,23 +1,30 @@
 //! Tests for displaying the cargo version.
 
-#![allow(deprecated)]
-
-use cargo_test_support::{cargo_process, project};
+use cargo_test_support::{cargo_process, project, str};
 
 #[cargo_test]
 fn simple() {
     let p = project().build();
 
     p.cargo("version")
-        .with_stdout(&format!("cargo {}\n", cargo::version()))
+        .with_stdout_data(str![[r#"
```

Comment:
It feel like the variable is important enough so should be kept with `.with_stdout(&format!("cargo {}\n", cargo::version()))` or similar. It also won't be a burden for manual updates, as this should be the only place we assert the version string.

See #14039 "4. Resolve non-literal deprecations"

Reply:
addressed at commit 5d44b52